pyproject.toml

```diff
@@ -81,7 +82,6 @@ dev = [
     "pre-commit==3.3.2",
     "pytest==7.1.2",
     "pytest-xdist==2.5.0",
-    "typeguard==2.13.3",
```

Comment:
> As `jaxtyping` depends on `typeguard` ([link](https://github.com/google/jaxtyping/blob/e347c480d51b2cf0671470b522ba09e38a0e9e25/pyproject.toml#L26)), move versioned `typeguard==2.13.3` from `dev` to base dependencies.

Do we use `typeguard` outside of development + automated tests? If not it makes sense to me to keep this pinned only in the `dev` dependencies.

Reply:
Some background - I ran into this issue trying to install dev while keeping base packages pinned to already installed versions.  If a user installs base nerfstudio, they get typeguard==4.3.1 (jaxtyping in the base requires typeguard>=2.13.3).  But if a user later installs dev nerfstudio, typeguard must be downgraded to 2.13.3 (unit tests don’t appear to pass with a newer version of typeguard). 

I found it unexpected for dev to introduce new version requirements for already installed base packages. Is that actually expected behavior?

Reply:
okie dokie - I removed the `typeguard` change and updated the PR name and description to the limited scope.

Reply:
Thanks!!